libraries/Microsoft.Bot.Builder.Ai/TranslationMiddleware.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Ai
+{
+    public class TranslationMiddleware : IMiddleware
+    {
+        private readonly string[] _nativeLanguages;
+        private readonly Translator _translator;
+        private readonly Dictionary<string,List<string>> _patterns;
+        private readonly Func<ITurnContext, string> _getUserLanguage;
+        private readonly Func<ITurnContext, Task<bool>> _setUserLanguage;
+
+        /// <summary>
+        /// Constructor for automatic detection of user messages
+        /// </summary>
+        /// <param name="nativeLanguages">List of languages supported by your app</param>
+        /// <param name="translatorKey"></param>
+        public TranslationMiddleware(string[] nativeLanguages, string translatorKey)
+        {
+            if(nativeLanguages==null || nativeLanguages.Count()==0)
```

Comment:
nativeLanguages.Length is better than .Count(). 

Using .Count()  uses linq, and is more complex and slower.

Reply:
Used Length instead of Count